model_analyzer/config/generate/perf_analyzer_config_generator.py

```diff
@@ -199,55 +246,148 @@ def _create_concurrency_list(self) -> List[int]:
-                    self._model_name, self._cli_config
+                new_perf_config = self._create_new_perf_config(
+                    parameter, unmodified_non_parameter_combination
                 )
+                all_perf_configs_for_a_given_parameter.append(new_perf_config)
 
-                new_perf_config.update_config(params)
+            self._configs.append(all_perf_configs_for_a_given_parameter)
 
-                if self._cli_config.is_request_rate_specified(self._model_parameters):
-                    new_perf_config.update_config({"request-rate-range": parameter})
-                else:
-                    new_perf_config.update_config({"concurrency-range": parameter})
+    def _create_new_perf_config(
+        self, parameter: int, unmodified_non_parameter_combination: Dict
+    ) -> PerfAnalyzerConfig:
+        perf_config = self._create_base_perf_config()
+
+        (
+            prompt_length,
+            modified_non_parameter_combination,
+        ) = self._extract_prompt_length(unmodified_non_parameter_combination)
+
+        self._update_perf_config_based_on_non_parameter_combination(
+            perf_config, modified_non_parameter_combination
+        )
+        self._update_perf_config_based_on_parameter(perf_config, parameter)
+        self._update_perf_config_based_on_perf_analyzer_flags(perf_config)
+        self._update_perf_config_for_llm_model(perf_config, prompt_length)
+
+        return perf_config
+
+    def _create_base_perf_config(self) -> PerfAnalyzerConfig:
+        perf_config = PerfAnalyzerConfig()
+        perf_config.update_config_from_profile_config(
+            self._model_name, self._cli_config
+        )
 
-                # User provided flags can override the search parameters
-                new_perf_config.update_config(self._perf_analyzer_flags)
+        return perf_config
 
-                configs_with_concurrency.append(new_perf_config)
-            self._configs.append(configs_with_concurrency)
+    def _extract_prompt_length(
+        self, unmodified_parameter_combination: Dict
+    ) -> Tuple[int, Dict]:
+        modified_parameter_combination = {
+            k: v for k, v in unmodified_parameter_combination.items()
+        }
+
+        if self._cli_config.is_llm_model():
+            prompt_length = modified_parameter_combination.pop("prompt-length")
+            return prompt_length, modified_parameter_combination
+        else:
+            return 0, unmodified_parameter_combination
+
+    def _update_perf_config_based_on_non_parameter_combination(
+        self, perf_config: PerfAnalyzerConfig, non_parameter_combination: Dict
+    ) -> None:
+        perf_config.update_config(non_parameter_combination)
+
+    def _update_perf_config_based_on_perf_analyzer_flags(
+        self, perf_config: PerfAnalyzerConfig
+    ) -> None:
+        perf_config.update_config(self._perf_analyzer_flags)
+
+    def _update_perf_config_based_on_parameter(
```

Comment:
I didn't realize we had a name for this! Before request-rate-range was introduced we just had batch size and concurrency. When I added in request-rate we had BS and {request-rate, concurrency}. I called this grouping with the "parameter". 
Then when we introduced the LLM fields we now had {BS, LMM} + {RR, C}, so I called the {BS, LLM} non-parameter.

I'll switch {RR, C} to be called inference_load. 
Any suggestions on what I should call the {batch_size, prompt-length, max-token-count} grouping?

Reply:
Don't have a good idea. Maybe {batch size, prompt length, max token, ...etc} could be called "parameters"? Since these are parameters to server/models and we no longer refer parameter to anything now.

Reply:
I've changed non-parameter to parameter